src/app/inventory/store/d2-item-factory.service.ts

```diff
@@ -903,6 +904,13 @@ export function D2ItemFactory(
       }
       const reusablePlugs = (socket.reusablePlugHashes || []).map((hash) => defs.InventoryItem.get(hash));
       const plugOptions = reusablePlugs.length > 0 && (!plug || !socket.plugHash || (socket.reusablePlugHashes || []).includes(socket.plugHash)) ? reusablePlugs : (plug ? [plug] : []);
+      // the merge is to enable the intrinsic mods to show up even if the user choosed another
+      // plug.itemTyper - removes the reusablePlugs from masterwork
+      // plug.action - removes the "Remove Shader" plug
+      if (reusablePlugs.length > 0 && plugOptions.length > 0) {
+        reusablePlugs.forEach((plug) => { if (!plugOptions.includes(plug) && plug.itemType && plug.action) { plugOptions.push(plug); } });
```

Comment:
Just a tip, you can use `for (const plug of reusablePlugs)` instead of `forEach`.